src/test/suite/logger.test.ts

```diff
@@ -0,0 +1,37 @@
+// import * as assert from 'assert';
+import * as vscode from 'vscode';
+import * as fs from 'fs';
+import * as os from 'os';
+import * as tmp from 'tmp';
+import * as assert from 'assert';
+import * as nisabaLogger from '../../logger';
+
+const nisabaOutputChannel = vscode.window.createOutputChannel("Nisaba");
+
+// Create temporary file and initialise logger with it.
+const tmpobj = tmp.fileSync();
+nisabaLogger.initLogging(nisabaOutputChannel, tmpobj.name);
+
+// Log messages to the temporary file.
+const messages = [
+    ['debug', 'This is a debug message'],
+    ['info', 'This is an info'],
+    ['warn', 'This is a warning'],
+    ['error', 'This is an error']
+]
+for (let i = 0; i < messages.length; i++) {
+    nisabaLogger.log(messages[i][0], messages[i][1]);
+}
```

Comment:
Not crucial, but we can also use a more functional interface with arrays:
```suggestion
messages.forEach(message => nisabaLogger.log(message[0], message[1]))
```

or even
```suggestion
messages.forEach(([level, message]) => nisabaLogger.log(level, message))
```